hub/api/dataset.py

```diff
@@ -35,8 +35,7 @@ def __init__(
         Args:
             path (str): The location of the dataset. Used to initialize the storage provider.
             mode (str): Mode in which the dataset is opened.
-                Supported modes include ("r", "w", "a") plus an optional "+" suffix.
```

Comment:
I'm fine with this but any particular reason why we're not keeping "w"?

Reply:
definitely makes sense to remove "w" actually in favor of "a", however i wonder if users expect this?

Reply:
@AbhinavTuli @McCrearyD If you open a file with "w", it deletes the existing file and starts a new one. I didn't know how to implement "delete this dataset" so I removed it for now

Reply:
we have Dataset.delete already that can be used

Reply:
I really don't think we want to do that. Is it really safe to delete the entire dataset when they do `Dataset(..., mode='w')`? I don't like the idea of supporting such a simple shorthand for something so dangerous. `mode='w'` really only makes sense over `mode='a'` when you want to make a new dataset, but I still think it's too easy to stumble onto this from other angles.